clvm/CLVMObject.py

```diff
@@ -1,6 +1,22 @@
+from __future__ import annotations
+
 import typing
 
 
+class CLVMStorage(typing.Protocol):
+    # It's not clear if it is possible to express the exclusivity without maybe
+    # restructuring all the classes, such as having a separate instance for each
+    # of the atom and pair cases and hinting a union of a protocol of each type.
+    atom: typing.Optional[bytes]
+    pair: typing.Optional[PairType]
```

Comment:
https://github.com/Chia-Network/clvm_rs/blob/6b3d0b5b2bb7cf84b61be586b7ec8c991d043b86/wheel/python/clvm_rs/clvm_storage.py#L11-L16
```python
class CLVMStorage(Protocol):
    atom: Optional[bytes]

    @property
    def pair(self) -> Optional[Tuple["CLVMStorage", "CLVMStorage"]]:
        ...
```

Is there a reason that `.pair` is read only?  Or that `.atom` is not?  I think there's a place in the tests that we actually mutate this, but I'd have to double check.

Reply:
This `CLVMStorage` protocol is a relatively recent addition to `clvm` with the goal to be more rigorous about type-checking. But it did get to the point of being widely used. There's a good chance we could actually just take it out without affecting too much.

We could make `.atom` a property to enforce immutability too, but that would be at the cost of performance. For all but the simplest cases where the entire tree is already encoded with the correct types, `.pair` is going to need to do some work, so making it a function call is okay.